WorkloadManagementSystem/scripts/dirac-wms-match.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+"""
+  Get computing resources capable to execute the given job
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC.Core.Base import Script
+from DIRAC import S_OK, gLogger, exit as DIRACExit
+
+fullMatch = False
+sites = None
+
+
+def setFullMatch(optVal_):
+  global fullMatch
+  fullMatch = True
+  return S_OK()
+
+
+def setSites(optVal_):
+  global sites
+  sites = optVal_.split(',')
+  return S_OK()
+
+
+description = """Get computing resources capable to execute a job with the given description.
+
+Note that only statically defined computing resource parameters are considered although sites
+can fail matching due to their dynamic state, e.g. occupancy by other jobs.
+"""
+
+Script.setUsageMessage('\n'.join([description,
+                                  'Usage:',
+                                  '%s [option]... <job_JDL>' % Script.scriptName]))
+
+Script.registerSwitch("F", "full-match", "Check all the matching criteria", setFullMatch)
+Script.registerSwitch("S:", "site=", "Check matching for these sites (comma separated list)", setSites)
+
+Script.parseCommandLine(ignoreErrors=True)
+args = Script.getPositionalArgs()
+
+if len(args) == 0:
+  gLogger.error("Error: No job description provided")
+  Script.showHelp()
+  DIRACExit(-1)
+
+from DIRAC.Core.Security.ProxyInfo import getVOfromProxyGroup
+from DIRAC.ConfigurationSystem.Client.Helpers import Resources
+from DIRAC.Core.Utilities.PrettyPrint import printTable
+from DIRAC.ResourceStatusSystem.Client.ResourceStatus import ResourceStatus
+from DIRAC.ResourceStatusSystem.Client.SiteStatus import SiteStatus
+from DIRAC.WorkloadManagementSystem.Utilities.QueueUtilities import getQueuesResolved, matchQueue
+
+
+if __name__ == '__main__':
+  with file(args[0]) as f:
```

Comment:
`file` doesn't exist in Python 3:
```suggestion
  with open(args[0]) as f:
```